### ISSUE TITLE
Error when converting tensorflow .pkl checkpoint to pytorch .pt checkpoint

### ISSUE THREAD

**Issue by kirkegross** [JCBrouwer/maua-stylegan2#13]
Hi!
I loaded in my own .pkl model and got this error after trying to convert tensorflow .pkl checkpoint to pytorch .pt checkpoint.
Is there something wrong with my model? 

Traceback (most recent call last):
  File "../stylegan2-pytorch/convert_weight.py", line 236, in <module>
    generator, discriminator, g_ema = pickle.load(f)
TypeError: 'Sequential' object is not iterable

Best,
Kirke

**Comment by JCBrouwer** [comment]
Hey there, it looks like the pickle file you're loading isn't a return a tuple of (generator, discriminator, g_ema).

It's probably only the generator (as that's all you need for the colab), so you could edit that line to remove ,discriminator,g_ema and reload the file.

**Comment by kirkegross** [comment]
Thank you, will try it out !

**Comment by kirkegross** [comment]
<img width="893" alt="Screenshot 2021-02-13 at 19 21 15" src="https://user-images.githubusercontent.com/23554239/107859323-a7f0e500-6e30-11eb-9029-cadf27027ccd.png">

If I removed this I got this error now. Can I also change this line to something else:
 size = g_ema.output_shape[2]

**Comment by JCBrouwer** [comment]
Oh, my bad `g_ema` is the one that gets used later on (that's the moving average of the generator which tends to have higher quality outputs). You should make sure the line is `g_ema = pickle.load(f)`.

**Comment by kirkegross** [comment]
Thank you! I changed line 236 to this g_ema = pickle.load(f)

<img width="1078" alt="Screenshot 2021-02-14 at 09 29 36" src="https://user-images.githubusercontent.com/23554239/107873037-2a1bf080-6ea7-11eb-8c51-0bfd96c8062f.png">

Now I got this:  AttributeError: 'Sequential' object has no attribute '_inbound_nodes'

**Comment by JCBrouwer** [comment]
How have you created this .pkl file? What repository did you use to train your StyleGAN2 network?

This error seems to show that your .pkl file doesn't have some properties that the correct tensorflow network checkpoint should have.

**Comment by kirkegross** [comment]
Ok I see, thanks! That's what I thought. I trained my model following this tutorial https://github.com/jeffheaton/t81_558_deep_learning/blob/master/t81_558_class_07_2_Keras_gan.ipynb
(Implementing DCGANs in Keras). In this tutorial the model was saved as .h5 and I converted it to .pkl. 

But I understand now that I should have trained my model as StyleGAN2 network. Thanks for all the help! I will train my model as StyleGAN2 now and then I can use this repo.
